src/CMakeLists.txt

```diff
@@ -19,8 +28,13 @@ set(lpcnet_freedv_srcs
     ${lpcnet_SOURCE_DIR}/nnet_data.c
 )
 
-add_library(lpcnetfreedv SHARED ${lpcnet_freedv_srcs})
-target_link_libraries(lpcnetfreedv codec2)
+# Set visibility of files brought in from Codec2 so that they're not visible to 
+# users (such as Codec2 itself).
+foreach(CODEC2_FILE IN LISTS codec2_import_srcs)
+    set_source_files_properties(${CODEC2_FILE} PROPERTIES COMPILE_FLAGS -fvisibility=hidden)
+endforeach()
+
```

Comment:
Gosh that's neat - I didn't know there was a way of looping in `cmake` :thinking: 

Well done on finding the magic `-fvisibility=hidden` linker option :+1: